src/pages/saksbehandling/steg/beregning/FradragInputs.tsx

```diff
@@ -0,0 +1,290 @@
+    feil: string | undefined;
+}) => (
+    <div>
+        <h3>{props.tittel}</h3>
+        <span className={styles.inputOgtekstContainer}>
+            <Input
+                className={styles.inputWithFollowTextInputfelt}
+                name={props.inputName}
+                value={props.value}
+                bredde={props.bredde}
+                onChange={props.onChange}
+            />
+            <Normaltekst>{props.inputTekst}</Normaltekst>
+        </span>
+        <Feilmelding>{props.feil ?? ''}</Feilmelding>
+    </div>
+);
+
+const FradragsSelection = (props: {
+    label: string;
+    id: string;
+    className: string;
+    name: string;
+    value: string;
+    onChange: (event: React.ChangeEvent<HTMLSelectElement>) => void;
+    feil: string | undefined;
+    intl: IntlShape;
+    optionLabel: string;
+}) => (
+    <div>
+        <h3>{props.label}</h3>
+        <Select
+            onChange={props.onChange}
+            id={props.id}
+            name={props.name}
+            value={props.value}
+            className={props.className}
+        >
+            <option value="">{props.intl.formatMessage({ id: 'fradrag.type.emptyLabel' })}</option>
+            {Object.values(Fradragstype).map((f) => (
+                <option value={f} key={f}>
+                    {props.intl.formatMessage({ id: fradragstypeResourceId(f) })}
+                </option>
+            ))}
+        </Select>
+        <Feilmelding>{props.feil ? props.feil : ''}</Feilmelding>
+    </div>
+);
+
+const fraUtlandInntekt = yup
+    .object<FraUtlandInntektFormData>()
+    .defined()
+    .when('fraUtland', {
+        is: true,
+        then: yup.object<FraUtlandInntektFormData>({
+            beløpUtenlandskValuta: validateStringAsNumber,
+            valuta: yup.string().required(),
+            kurs: validateStringAsNumber,
+        }),
+        otherwise: yup.object<FraUtlandInntektFormData>(),
+    });
+
+const delerAvPeriode = yup
+    .object<DelerAvPeriodeFormData>()
+    .defined()
+    .when('delerAvPeriodeChecked', {
+        is: true,
+        then: yup.object<DelerAvPeriodeFormData>({
+            fraOgMed: yup.date().required().nullable(),
+            tilOgMed: yup.date().required().nullable(),
+        }),
+        otherwise: yup.object<DelerAvPeriodeFormData>(),
+    });
+
+export const fradragSchema = yup.object<FradragFormData>({
+    beløp: validateStringAsNumber,
+    type: yup.string().defined().oneOf(Object.values(Fradragstype), 'Du må velge en fradragstype'),
+    fraUtland: yup.boolean(),
+    fraUtlandInntekt: fraUtlandInntekt,
+    delerAvPeriodeChecked: yup.boolean(),
+    delerAvPeriode: delerAvPeriode,
+});
+
+export const isValidFradrag = (f: unknown): f is Fradrag => fradragSchema.isValidSync(f);
+
+export const FradragInputs = (props: {
+    fradrag: Array<FradragFormData>;
+    feltnavn: string;
+    errors: string | string[] | FormikErrors<FradragFormData>[] | undefined;
```

Comment:
Att errors kan vara så många olika typer gör det lite vanskeligt att jobba med. Jag ser längre ned att det blir en del type-checking på variabler för att finna ut vad man ska göra. Kanske hade varit digg att komma runt det om det är möjligt? 😊

Reply:
Jeg er usikker på hvordan denne kan løses. errors objectet kan være dem typene og er egentlig bare det vi passer nedover. Skal ta en runde på det også :)

Reply:
Dette blir litt knotete som dette fordi `formik` sine feil kan være hva som helst av dette (ifølge typingen iallfall`. Tittet litt på det for en stund siden, men kom ikke til noen veldig god generell måte å forenkle det på :/